hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationPlanner.java

```diff
@@ -115,23 +116,22 @@ void planMigrations(Address[] oldAddresses, Address[] newAddresses, MigrationDec
                                 + ", CURRENT: " + Arrays.toString(state) + ", FINAL: " + Arrays.toString(newAddresses));
             }
 
-            if (newAddresses[currentIndex].equals(state[currentIndex])) {
-                // No change, no action needed.
+            if (target.equals(state[currentIndex])) {
+                // no change, no action needed
                 currentIndex++;
                 continue;
             }
 
-            if (getReplicaIndex(newAddresses, state[currentIndex]) == -1
-                    && getReplicaIndex(state, newAddresses[currentIndex]) == -1) {
+            if (getReplicaIndex(newAddresses, state[currentIndex]) == -1 && getReplicaIndex(state, target) == -1) {
                 // MOVE partition replica from its old owner to new owner
-                log("MOVE %s to index: %d", newAddresses[currentIndex], currentIndex);
-                callback.migrate(state[currentIndex], currentIndex, -1, newAddresses[currentIndex], -1, currentIndex);
-                state[currentIndex] = newAddresses[currentIndex];
+                log("MOVE %s to index: %d", target, currentIndex);
+                callback.migrate(state[currentIndex], currentIndex, -1, target, -1, currentIndex);
+                state[currentIndex] = target;
```

Comment:
IMO, reading `state[currentIndex] = newAddresses[currentIndex]` easier, because seeing an indexed read/write is better than to remember what `target` was.

Reply:
Fixed. I also applied this to the pulled out method, so we have the same style there.